src/platform/plugins/shared/workflows_execution_engine/server/workflow_context_manager/workflow_execution_state.ts

```diff
@@ -19,6 +19,7 @@ export class WorkflowExecutionState {
   private workflowExecution: EsWorkflowExecution;
   private workflowDocumentChanges: Partial<EsWorkflowExecution> | undefined = undefined;
   private stepDocumentsChanges: Map<string, Partial<EsWorkflowStepExecution>> = new Map();
+  private variables: Record<string, unknown> = {};
```

Comment:
I'm afraid this will not survive after wait step or a step that resumes execution in another task.

Reply:
@skynetigor what's the difference from steps then? can you guide me through where we should store and reload it after resume?